### ISSUE TITLE
Task 4.1: Ethical Training Dataset Generation

### ISSUE THREAD

**Issue by nfriacowboy** [nfriacowboy/ethical-snn-research#10]
**Priority**: 🔴 Critical  
**Status**: ⏸️ Not started  
**Estimated time**: 6 hours  
**Week**: 4 - Ethical SNN & Dual-Process Integration

## Subtasks
- [ ] Create `scripts/generate_ethical_dataset.py`
  - [ ] Generate 1000 synthetic scenarios
  - [ ] Format: (state, action, is_ethical: bool)
  - [ ] Balance classes (~50/50 ethical/unethical)
  - [ ] Save to `data/ethical_training/scenarios.json`
- [ ] Create `src/training/ethical_dataset.py`
  - [ ] Class `EthicalDataset(torch.utils.data.Dataset)`
  - [ ] Method: `__getitem__` returns (state_tensor, label)
- [ ] Add unit tests: `tests/test_ethical_dataset.py`

## Acceptance Criteria
```bash
python scripts/generate_ethical_dataset.py
# Creates scenarios.json with 1000 entries
```

**Dependencies**: Task 3.1  
**Blocks**: Task 4.2
